src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs

```diff
@@ -779,9 +779,9 @@ private async Task<HttpConnectionContext> GetOrCreateConnectionAsync(HttpContext
 
         private HttpConnectionContext CreateConnection(HttpConnectionDispatcherOptions options)
         {
+            _manager.DisconnectTimeout = options.DisconnectTimeout;
```

Comment:
You cannot make the disconnect timeout a per-endpoint option and then store it in the singleton HttpConnectionManager. This will cause the HttpConnectionManager to enforce the most recent connection's disconnect timeout for *all* connections including those that might belong to a differently-configured endpoint.

Reply:
Good catch. Hmm what would be the best way to do this then

Reply:
I'll make an HttpConnectionManagerOption type and configure it via an `addSignalR` overload.